packages/store/src/action.ts

```diff
@@ -31,12 +13,12 @@ export function Action(actions: any | any[], options?: ActionOptions) {
     }
 
     for (const action of actions) {
+      const type = action.type;
+
       if (!action.type) {
-        action.type = new ActionToken(action.name);
+        throw new Error(`Action ${action.name} is missing a static "type" property`);
       }
```

Comment:
since the compile time error wasn't as helpful as we would like I added a small runtime check